binaries/cli/src/main.rs

```diff
@@ -227,7 +259,65 @@ fn run() -> eyre::Result<()> {
             }
         }
         Command::Destroy { config } => up::destroy(config.as_deref())?,
-    }
+        Command::Coordinator { port } => {
+            let rt = Builder::new_multi_thread()
+                .enable_all()
+                .build()
+                .context("tokio runtime failed")?;
+            rt.block_on(async {
+                let (_, task) = dora_coordinator::start(
+                    port,
+                    ctrlc_events.map(|event| match event {
+                        Event::CtrlC => dora_coordinator::Event::CtrlC,
+                    }),
+                )
+                .await?;
+                task.await
+            })
+            .context("failed to run dora-coordinator")?
+        }
+        Command::Daemon {
+            coordinator_addr,
+            machine_id,
+            run_dataflow,
+        } => {
+            let rt = Builder::new_multi_thread()
+                .enable_all()
+                .build()
+                .context("tokio runtime failed")?;
+            rt.block_on(async {
+                match run_dataflow {
+                    Some(dataflow_path) => {
+                        tracing::info!("Starting dataflow `{}`", dataflow_path.display());
+
+                        Daemon::run_dataflow(&dataflow_path).await
+                    }
+                    None => {
+                        Daemon::run(
+                            coordinator_addr.unwrap_or_else(|| {
+                                tracing::info!("Starting in local mode");
+                                let localhost = Ipv4Addr::new(127, 0, 0, 1);
+                                (localhost, DORA_COORDINATOR_PORT_DEFAULT).into()
+                            }),
+                            machine_id.unwrap_or_default(),
+                            ctrlc_events.map(|event| {
+                                let clock = HLC::default();
+                                match event {
+                                    Event::CtrlC => Timestamped {
+                                        inner: dora_daemon::Event::CtrlC,
+                                        timestamp: clock.new_timestamp(),
```

Comment:
Note that the timestamp will be less exact this way. I'm not sure whether this might lead to any issue in the future though, so I'm fine keeping it this way. Maybe add a short line comment that this timestamp is delayed a bit...